exporters/trace/stackdriver/src/main/java/io/opencensus/exporter/trace/stackdriver/StackdriverV2ExporterHandler.java

```diff
@@ -171,15 +173,42 @@ static StackdriverV2ExporterHandler createWithStub(
   static StackdriverV2ExporterHandler createWithCredentials(
       String projectId,
       Credentials credentials,
-      Map<String, io.opencensus.trace.AttributeValue> fixedAttributes)
+      Map<String, io.opencensus.trace.AttributeValue> fixedAttributes,
+      @javax.annotation.Nullable Duration deadline)
       throws IOException {
-    TraceServiceSettings traceServiceSettings =
+    TraceServiceSettings.Builder builder =
         TraceServiceSettings.newBuilder()
             .setCredentialsProvider(
-                FixedCredentialsProvider.create(checkNotNull(credentials, "credentials")))
-            .build();
+                FixedCredentialsProvider.create(checkNotNull(credentials, "credentials")));
+    if (deadline != null) {
+      // We only use the batchWriteSpans API in this exporter.
+      builder
+          .batchWriteSpansSettings()
+          .setSimpleTimeoutNoRetries(org.threeten.bp.Duration.ofMillis(deadline.toMillis()));
+    } else {
+      /*
+       * Default retry settings for Stackdriver Trace client is:
+       * settings =
+       *   RetrySettings.newBuilder()
+       *       .setInitialRetryDelay(Duration.ofMillis(100L))
+       *       .setRetryDelayMultiplier(1.2)
+       *       .setMaxRetryDelay(Duration.ofMillis(1000L))
+       *       .setInitialRpcTimeout(Duration.ofMillis(30000L))
+       *       .setRpcTimeoutMultiplier(1.5)
+       *       .setMaxRpcTimeout(Duration.ofMillis(60000L))
+       *       .setTotalTimeout(Duration.ofMillis(120000L))
+       *       .build();
+       *
+       * Override the default settings with settings that don't retry.
+       */
+      builder
```

Comment:
So, iiuc, after this change there will be no retry for exporting traces and they will be dropped unless users set an explicit deadline, at which point they will get the default retry settings - is that about right?

Reply:
Partially correct, when users set an explicit deadline still no retries will be performed (`setSimpleTimeoutNoRetries`). It's the expected behavior that not to do retry by default, since by doing retries we hold references to undelivered spans longer, thus it makes memory usage worse. Note users always have an option to use a custom Stackdriver Trace client: https://github.com/census-instrumentation/opencensus-java/blob/4475dd9ec123e0378bcb4726fc261e6fe233daf4/exporters/trace/stackdriver/src/main/java/io/opencensus/exporter/trace/stackdriver/StackdriverTraceConfiguration.java#L127, which they can configure to use other retry settings.

Maybe we can also consider adding a top-level `retrySettings` option but I'd prefer to wait and see if it's really needed.

Reply:
Got it - I think that's a good approach.